packages/server-wallet/src/protocols/process-ledger-queue.ts

```diff
@@ -30,27 +31,42 @@ export type ProtocolState = {
   channelsReturningFunds: ChannelState[];
 };
 
+function removeChannelFromAllocation(
+  allocationItems: AllocationItem[],
+  channel: ChannelState
+): AllocationItem[] {
+  if (!channel.supported) throw new Error('state is unsupported');
+
+  const {allocationItems: channelAllocations} = checkThat(
+    channel.supported.outcome,
+    isSimpleAllocation
+  );
+
+  const [removed, remaining] = _.partition(allocationItems, ['destination', channel.channelId]);
+
+  if (removed.length !== 1) throw 'Expected to find exactly one item';
+
+  if (removed[0].amount !== channelAllocations.map(x => x.amount).reduce(BN.add, BN.from(0)))
+    throw 'Bookkeeping error';
```

Comment:
Can we use a more appropriate error? I did not intend for this message to be used verbatim, it was meant as a placeholder.

Reply:
We don't really have a known error type for this, so I might just keep it for now. I suppose it would fit inside some category of "protocol errors"?

Reply:
I just mean `'Bookkeeping error'` is not very informative, but maybe it's fine. It should at least be `Error('Bookkeeping error')` though.